DOCSP-50809: bson 5.1 release (#159)
mongodb/docs-ruby: source/reference/release-notes.txt

- Adds compatibility with {+language+} 3.3.

BSON v5.1
~~~~~~~~~

The ``bson`` gem is the implementation of the BSON specification in
{+language+}. ``bson`` v5.1 accompanies the {+driver-short+} v2.22
release.

The ``bson`` 5.1 release includes the following new features, changes,
and improvements:

- Adds the ``BSON::Vector`` type to represent vectors of numeric values.
  To learn more, see the :ref:`ruby-bson-vector` section of the BSON
  guide.

> - The ``#deep_symbolize_keys!`` and ``#symbolize_keys!`` methods are not
>   supported on ``BSON::Document`` objects. In previous versions,

Reviewer: Reword to avoid passive voice

```suggestion
- ``BSON::Document`` objects no longser support the ``#deep_symbolize_keys!`` and ``#symbolize_keys!`` methods. In previous versions,
```

- Adds compatibility with {+language+} 3.3.

BSON v5.1
~~~~~~~~~

The ``bson`` gem is the implementation of the BSON specification in
{+language+}. ``bson`` v5.1 accompanies the {+driver-short+} v2.22

> release.

Reviewer 2: We actually discovered that all the changes we meant to release in driver v2.22 somehow got released in 2.21.1. This includes the vector search support in the driver, though it wouldn't have worked until bson 5.1.0.

So, bson 5.1.0 supports ruby driver 2.21.1. There will be a driver 2.22 eventually, but probably not imminently.
